src/hooks/use-apps.tsx

```diff
@@ -1,22 +1,28 @@
+import { useCallback, useEffect, useState } from 'react'
 import { AppService } from '@uxshop/storefront-core'
 import { AppFields } from '@uxshop/storefront-core/dist/modules/app/AppTypes'
+import { HookData } from './types/HookData'
 
 interface AppHookParams {
   id: string
 }
 
 export function useApps({ id }: AppHookParams, fields?: AppFields[]): any {
-  function getById({ id }: AppHookParams, fields?: AppFields[]) {
-    let result = {
-      data: null,
-      error: null
-    }
+  const [state, setState] = useState<HookData>({
+    loading: false,
+    data: null
+  })
+
+  const getById = useCallback(() => {
+    setState({ ...state, loading: true })
```

Comment:
@kevenssousa não é melhor fazer com que o state de loading já comece como true inicialmente? Assim, não precisa fazer um set a mais, e se der algum erro ou se o fluxo for concluído, aí sim setar pra false. @simaomeyrerdooca o que tu acha?

Reply:
As duas abordagens funcionam, teoricamente só as funções internas dos hooks são as que deveriam setar o loading, acabamos mantendo assim por que foram os primeiros que fizemos nessa estrutura, nesse caso teria que avaliar mais a fundo e ver se vale a pena a troca em todos os hooks mas devido a entrega podemos manter assim nesse primeiro momento